src/mbgl/map/transform.cpp

```diff
@@ -313,6 +313,107 @@ void Transform::_easeTo(const CameraOptions& options, double new_scale, double n
     }
 }
 
+void Transform::flyTo(const CameraOptions &options) {
```

Comment:
I would love to, but a lot of this stuff I don't fully understand myself; I inherited it from @1ec5's branch, and he inherited it from the GL-JS world.  It's not actually clear who originally wrote some of this math.  I agree this is not an ideal situation.